client/my-sites/checkout/cart/cart-coupon.jsx

```diff
@@ -156,4 +164,26 @@ export class CartCoupon extends React.Component {
 	};
 }
 
-export default localize( CartCoupon );
+const mapDispatchToProps = dispatch => ( {
+	recordGoogleEvent,
+	applyCoupon: coupon =>
+		dispatch(
+			withAnalytics(
+				recordTracksEvent( 'calypso_checkout_coupon_submit', {
+					coupon_code: coupon,
+				} ),
+				() => applyCoupon( coupon )
```

Comment:
Do these play well together? It looks like `applyCoupon()` here is a function that typically wouldn't be dispatched. If I'm mistaken then you can ignore me. However, can you verify that this actually has its effect when dispatched? I guess the idea is that we're dispatching a thunk which isn't a Redux action which then runs?

Maybe simpler could be this…

```js
applyCoupon: coupon_code => {
	dispatch( recordTracksEvent( 'calypso_checkout_coupon_submit', { coupon_code } ) );
	applyCoupon( coupon );
}
```

Reply:
Good suggestion. You're right. These were working, but it's also not necessary to dispatch them because internally they call `Dispatcher.handleViewAction()` on their own. I simplified both `applyCoupon()` and `removeCoupon()` using your idea.